### ISSUE TITLE
Feature Request: `parse_pipfile` function

### ISSUE THREAD

**Issue by malikoth** [di/pip-api#47]
With adoption increasing for `Pipfile` based dependency declaration, it would be great if this library could parse a `Pipfile` as well.  Nice thing is that since a `Pipfile` doesn't support nested files like `requirements.txt` does, and the format of the file is just TOML, it should be a fairly short function that just reuses a lot of the existing functionality.

If this seems reasonable scope for this project, I'll happily submit a PR with it.  :)

**Comment by di** [comment]
Hi @malikoth, thanks for the feature request.

The purpose of this package to providing a drop-in replacement for existing uses of `pip`'s internal API by wrapping command-line calls to `pip`. To my knowledge, `pip` doesn't currently support parsing a `Pipfile` so there is no equivalent function or class one could import from `pip` which provides this functionality, so there's nothing for us to replace or wrap.

I think if `pip` ever begins supporting installing from a `Pipfile`, there probably would be a need for something like `pip_api.parse_pipfile`, but until then there's not much to do here I think.

If you just want to be able to parse a `Pipfile`, I think this project is probably the wrong place for that functionality, and maybe https://pypi.org/project/pipfile/ would make more sense.

**Comment by malikoth** [comment]
Fair enough.  I'd been thinking that `parse_requirements` was more of a feature of this project than a wrapper around pip itself; my bad.

That being said, the `Pipfile` project does say

> Pip Integration (eventual)
++++++++++++++++++++++++++
`pip`_ will grow a new command line options, ``-p`` / ``--pipfile`` to install the versions as specified in a ``Pipfile``, similar to its existing ``-r`` / ``--requirement`` argument for installing ``requirements.txt`` files.

In that hypothetical eventual future, maybe we will be adding this in the future.  :)

**Comment by di** [comment]
Yep! I expect `pip` to gain this support eventually, so I'll leave this issue open until then.

**Comment by malikoth** [comment]
Or feel free to close, and reopen when / if appropriate.
